### ISSUE TITLE
Bug: The selected text highlight often disappears on right-click.

### ISSUE THREAD

**Issue by kmgallahan** [microsoft/microsoft-ui-xaml#1801]
**Describe the bug**

The selected text highlight disappears most of the time after right clicking it. This applies to:

- TextBox
- TextBlock
- RichTextBlock
- RichEditBox
- PasswordBox
- AutoSuggestBox

**Steps to reproduce the bug**

1. Open the XAML Controls Gallery > Text > any sample
2. Select some text (enter it first if necessary)
3. Right-click said text to bring up the context menu

**NOTE**: It doesn't always happen on the first right-click, but simply right-click again on the text without closing the context menu and it will typically disappear then.

**Expected behavior**

The selected text highlight should remain visible until the selected text is no longer selected.

**Contrast this behavior to right-clicking text in OneNote**; you can right-click selected text there as many times as you like and whatever was selected will keep its highlight the whole time.

**Version Info**
XAML Controls Gallery v.1.2.11.0

NuGet package version: 

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2019 Update (18362)            | Yes  |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 |  Yes/No? |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Comment by ranjeshj** [comment]
I see this on controls gallery as well.

**Comment by yaichenbaum** [comment]
I had this issue in one of my apps and I fixed it with this, I hope this helps 🙂

```
<TextBox>
  <TextBox.SelectionHighlightColorWhenNotFocused>
    <SolidColorBrush Color="{StaticResource SystemAccentColor}"/>
  </TextBox.SelectionHighlightColorWhenNotFocused>
</TextBox>
```

**Comment by kmgallahan** [comment]
@yaichenbaum 

Using `SelectionHighlightColorWhenNotFocused` is just a workaround rather than a fix, as now when switching apps your highlighted text can't change color.

Popping a context menu for selected text shouldn't cause the text to lose focus. Or if it has to lose focus, then there needs to be logic or a new property to handle this scenario (although that property would/should be equal to `SelectionHighlightColor ` by default).

**Comment by chrisglein** [comment]
I can also repro. It does just seem to be the visual state; selection state restores when the menu closes. Given the workaround above it does seem like focus is getting lost/restored.
@didalp (Text) and @Austin-Lamb (Focus): Does this at all sound familiar (TextBox losing focus while context menu is up)?

**Comment by ddalp** [comment]
@chrisglein It is a bug. @marlenecota had to fake the focus still on Text Control while floatie is shown. This looks like another case we did not handle the fake focus well.

**Comment by kmgallahan** [comment]
Keep alive

**Comment by ranjeshj** [comment]
Seems to be working for me in WinUI 3 Gallery everywhere but RichTextBlock overflow region.

**Comment by JJBrychell** [comment]
There was fix for this a while back to took care of most of the cases, except for RichTextBlockOverflow.  When focus moves from one of these controls, it checks to see if the target of the focus (the popup) is owned by the text control and if not, it will remove the selection display.   However, in the case of RichTextBlockOverflow, it does not own the popup.  Its primary RichTextBlock does.  So the RichTextBlockOverflow was thinking that focus was moving to some other Text control and removing the selection.

Fix is in and scheduled for 1.5.
